### PR TITLE
Enhance Performance

### PR DESCRIPTION
**Pull request by bluefloyd00**
- Replaced asyncIO which requires too many changes in pygyver.etl.dw with threading and concurrent libraries

### CODE REVIEW

File: pygyver/etl/dw.py [hunk 1]
```diff
@@ -305,12 +305,10 @@ def create_table(self, table_id, dataset_id=bq_default_dataset(), sql=None, file
 
         if sql is None and file is None:
             raise BigQueryExecutorError("EIther sql or file must be provided")
+
         if sql is None:
             sql = read_sql(file, **kwargs)
 
-        if not self.dataset_exists(dataset_id=dataset_id):
```
Comment (KarimZaoui) [inline pygyver/etl/dw.py:311]: why?

Comment (bluefloyd00) [reply pygyver/etl/dw.py:311]: it's to handle the multithreading and the API latency, I added it some time ago and I am removing it now that the concurrent thread is in place: all the thread check whether the dataset exists (API call), all of them get the answer the dataset does not exist, all of them try to create the dataset (another API call), only the first API call create the dataset, all the others return an error

File: tests/entrypoint-tests.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ fi
 
 if [ "$1" == "local" ]; then
     echo "Starting pygyver-tests-local entrypoint"
-    pytest --junitxml='pygyver-out-waxit-tests.xml' -sv || [[ $? -eq 1 ]]
+    pytest tests/test_pipeline.py --junitxml='pygyver-out-waxit-tests.xml' -sv || [[ $? -eq 1 ]]
```
Comment (KarimZaoui) [inline tests/entrypoint-tests.sh:20]: to be removed

File: pygyver/etl/lib.py [hunk 3]
```diff
@@ -162,8 +162,8 @@ def apply_kwargs(orig, kwargs):
     if orig is not None:
         for key_, value_ in orig.items():
             for kwargs_key, kwargs_value in kwargs.items():
-                if '$' + kwargs_key == value_:
-                    orig.update({key_: kwargs_value})
+                if '$' + kwargs_key in value_:
+                    orig.update({key_: value_.replace('$' + kwargs_key, kwargs_value)})
```
[the remark this answers is not in the document]
Comment (bluefloyd00) [reply pygyver/etl/lib.py:166]: good point, I will make sure that it compares from the $ to the end of the string then

File: pygyver/etl/pipeline.py [hunk 4]
```diff
@@ -95,17 +96,33 @@ def __init__(self, yaml_file, dry_run=False, *args, **kwargs):
...
+                    )
+
+                # drop datasets 
+                if args_dataset != []:            
+                    execute_parallel(
+                                self.remove_dataset,
+                                args_dataset,
+                                message='delete dataset: ',  
+                                log='dataset_id'                      
+                                )
```
Comment (KarimZaoui) [inline pygyver/etl/pipeline.py:125]: This is where the problem is I think

**Pull request merged**